src/Profile_User.php

```diff
@@ -132,144 +132,106 @@ public static function showForUser(User $user)
+        $order       = strtoupper($_GET["order"] ?? "");
+        $sort_params = [];
+        if ($sort !== '') {
+            $sort_params = [$sort => $order === 'DESC' ? 'DESC' : 'ASC'];
         }
+        $iterator = self::getListForItem($user, $start, $limit, $sort_params);
+        $total_num = self::countForItem($user);
 
-        if ($num > 0) {
-            echo "<table class='tab_cadre_fixehov'>";
-            $header_begin  = "<tr>";
-            $header_top    = '';
-            $header_bottom = '';
-            $header_end    = '';
-            if ($canedit) {
-                $header_begin  .= "<th>";
-                $header_top    .= Html::getCheckAllAsCheckbox('mass' . __CLASS__ . $rand);
-                $header_bottom .= Html::getCheckAllAsCheckbox('mass' . __CLASS__ . $rand);
-                $header_end    .= "</th>";
+        $entries = [];
+        $entity_form_url = Toolbox::getItemTypeFormURL('Entity');
+        foreach ($iterator as $data) {
+            $entry = [
+                'itemtype' => __CLASS__,
+                'id'       => $data['linkid'],
+            ];
+            $link = $data["completename"];
+            if ($_SESSION["glpiis_ids_visible"]) {
+                $link = sprintf(__('%1$s (%2$s)'), $link, $data["entities_id"]);
             }
-            $header_end .= "<th>" . Entity::getTypeName(Session::getPluralNumber()) . "</th>";
-            $header_end .= "<th>" . sprintf(
-                __('%1$s (%2$s)'),
-                self::getTypeName(Session::getPluralNumber()),
-                __('D=Dynamic, R=Recursive')
-            );
-            $header_end .= "</th></tr>";
-            echo $header_begin . $header_top . $header_end;
-
-            foreach ($iterator as $data) {
-                echo "<tr class='tab_bg_1'>";
-                if ($canedit) {
-                    echo "<td width='10'>";
-                    if (in_array($data["entities_id"], $_SESSION['glpiactiveentities'])) {
-                        Html::showMassiveActionCheckBox(__CLASS__, $data["linkid"]);
-                    } else {
-                        echo "&nbsp;";
-                    }
-                    echo "</td>";
-                }
-                echo "<td>";
+            if ($canshowentity) {
+                $link = "<a href='" . $entity_form_url . "?id=" . $data["entities_id"] . "'>" . $link . "</a>";
+            }
+            $entry['entity'] = $link;
 
-                $link = $data["completename"];
-                if ($_SESSION["glpiis_ids_visible"]) {
-                    $link = sprintf(__('%1$s (%2$s)'), $link, $data["entities_id"]);
-                }
+            if (Profile::canView()) {
+                $form_url = Toolbox::getItemTypeFormURL('Profile');
+                $profile_name = "<a href='{$form_url}?id={$data['id']}'>" . Html::entities_deep($data['name']) . "</a>";
+            } else {
+                $profile_name = Html::entities_deep($data['name']);
```

Comment:
I think it is prefereable to use the native PHP function directly.
```suggestion
                $profile_name = htmlspecialchars($data['name']);
```